int32 starting_index
---
bool success